Source/WebGPU/WebGPU/CommandEncoder.swift

```diff
@@ -32,7 +37,314 @@ public func resolveQuerySet(commandEncoder: WebGPU.CommandEncoder, querySet: Web
 {
     commandEncoder.resolveQuerySet(querySet, firstQuery: firstQuery, queryCount: queryCount, destination: destination, destinationOffset: destinationOffset)
 }
+
+@_expose(Cxx)
+public func CommandEncoder_copyBufferToTexture_thunk(commandEncoder: WebGPU.CommandEncoder, source: WGPUImageCopyBuffer, destination: WGPUImageCopyTexture, copySize: WGPUExtent3D){
+    commandEncoder.copyBufferToTexture(source: source, destination: destination, copySize: copySize)
+}
+
 extension WebGPU.CommandEncoder {
+    public func copyBufferToTexture(source: WGPUImageCopyBuffer, destination: WGPUImageCopyTexture, copySize: WGPUExtent3D) {
+        guard source.nextInChain == nil && source.layout.nextInChain == nil && destination.nextInChain == nil else {
+            return
+        }
+        guard self.prepareTheEncoderState() else {
+            self.generateInvalidEncoderStateError()
+            return
+        }
+        let destinationTexture = WebGPU.fromAPI(destination.texture)
+
+        let error = self.errorValidatingCopyBufferToTexture(source, destination, copySize)
+        guard error != nil else {
+            self.makeInvalid(error)
+            return
+        }
+        let apiBuffer = WebGPU.fromAPI(source.buffer)
+        apiBuffer.setCommandEncoder(self, false) 
+        destinationTexture.setCommandEncoder(self)
+        guard !(copySize.width != 0 && copySize.height != 0 && copySize.depthOrArrayLayers != 0) else {
+            return
+        }
+        guard !(apiBuffer.isDestroyed() || destinationTexture.isDestroyed()) else {
+            return
+        }
```

Comment:
I think the logic here is wrong. It should be something like this:

```suggestion
        guard copySize.width != 0 || copySize.height != 0 || copySize.depthOrArrayLayers != 0,
            !apiBuffer.isDestroyed(),
            !destinationTexture.isDestroyed()
            else {
                return
        }
```

I think combining into one guard clause makes it more understandable, too.

Reply:
good catch!. I will fix that.